template/koa-rest-api/src/framework/validation.test.ts

```diff
@@ -60,13 +66,21 @@ describe('validate', () => {
       .expect(422)
       .expect(({ body }) =>
         expect(body).toMatchInlineSnapshot(`
-          {
-            "invalidFields": {
-              "/description": "Required",
-              "/id": "Required",
-            },
-            "message": "Input validation failed",
-          }
-        `),
+{
+  "invalidFields": {
+    "/description": [
+      "Required",
+    ],
+    "/id": [
+      "Required",
```

Comment:
Hmmmm. Is this very useful? Should union errors be grouped accordingly by the member they came from?

Reply:
We could duplicate the error message for each field.

However, the new implementation at least tell us which field is invalid compared to the old implementation 😛
check unit tests in [first commit](https://github.com/seek-oss/skuba/pull/1339/commits/c574dd142cb9b99a4777c9e47318f26977286138)(only union schema changes)

Reply:
What I'm suggesting is instead of aggregating them, could we perhaps split them into:

```
union1: { (union 1 errors }
union2: { union 2 errors }
```
At the moment I'm not sure what combination of fields would make it not error

Reply:
I don't know if that is possible. 
The `ZodError` does not provide which union it comes from.

Reply:
Don't you have an array of union errors you're mapping over?

https://github.com/seek-oss/skuba/pull/1339/files#diff-4d7c9e9cdc6b4d6ed3ee02ed3b4a0c5aa141aabfa04ab2169909f9fe269ccca4R39

Reply:
Updated with the union index, please have a look.